src/Ryujinx.Graphics.Gpu/Image/AutoDeleteCache.cs

```diff
@@ -56,6 +61,25 @@ class AutoDeleteCache : IEnumerable<Texture>
 
         private readonly Dictionary<TextureDescriptor, ShortTextureCacheEntry> _shortCacheLookup;
 
+        /// <summary>
+        /// Initializes the cache, setting the maximum texture capacity for the specified GPU context.
+        /// </summary>
+        /// <remarks>
+        /// If the backend GPU has 0 memory capacity, the cache size defaults to `DefaultTextureSizeCapacity`.
+        /// </remarks>
+        /// <param name="context">The GPU context that the cache belongs to</param>
+        public void Initialize(GpuContext context)
+        {
+            var CacheMemory = (ulong)(context.Capabilities.MaximumGpuMemory * MemoryScaleFactor);
+
+            _maxCacheMemoryUsage = Math.Clamp(CacheMemory, MinTextureSizeCapacity, MaxTextureSizeCapacity);
+
+            if (CacheMemory == 0)
```

Comment:
```suggestion
            if (context.Capabilities.MaximumGpuMemory == 0)
```
I'd use that directly here. It's a bit weird to use the divided value.